src/components/RangeDatePicker/BaseRangeDatePicker.tsx

```diff
@@ -0,0 +1,285 @@
+import React from 'react'
+import { CSSProperties, Ref, useEffect, useRef, useState } from 'react'
+
+import { ExternalStyles, focusBoxShadow, Theme, useStyles } from '../../styles'
+import { composeRefs } from '../../util/react'
+import { DateInput } from '../DateField'
+import { Icons } from '../Icon'
+import { InputWrapper } from '../TextField/InputWrapper'
+
+export interface Period {
+  startDate?: Date
+  finalDate?: Date
+}
+
+export interface BaseRangeDatePickerProps {
+  /**
+   * Prop used to set an initial period.
```

Comment:
I believe using the term "prop" here is a little redundant, since all these attributes are already known to be the component props. Maybe just "The current component period value" in this case